### PR TITLE
[Bridges] add unified testing infrastructure

### PR DESCRIPTION
**Pull request by odow**
This is the start of an exploration to see if we can simplify how we test bridges. 

Alternative to #820
Alternative to #750 

Things that aren't addressed:
 - [x] Objective bridges
 - [ ] ConstraintPrimal
 - [ ] ConstraintDual
 - [x] ConstraintPrimalStart
 - [x] ConstraintDualStart
 - [x] Deletion

Out of scope
 - Variable bridges
 - VariablePrimalStart
 - Modifications
 - Updates

### CODE REVIEW

**Comment by odow** [comment]
This is turning out to be pretty useful. It found a bunch of problems with constraint primal/dual starts in the bridges.

File: src/Bridges/Constraint/bridges/geomean.jl [hunk 1]
```diff
@@ -218,9 +218,9 @@ function MOI.get(
 end
 
 function MOI.get(
-    b::GeoMeanBridge{T,F,G},
-    ::MOI.NumberOfConstraints{G,MOI.Nonnegatives},
-)::Int64 where {T,F,G}
+    b::GeoMeanBridge{T,F,G,H},
+    ::MOI.NumberOfConstraints{H,MOI.Nonnegatives},
+)::Int64 where {T,F,G,H}
```
Comment (odow) [inline src/Bridges/Constraint/bridges/geomean.jl:223]: These tests found a number of nasty bugs like this one!

**Comment by odow** [comment]
I'm a bit stuck on how to test constraint primal/dual. 

Any ideas? We test the start values by round-tripping them. It doesn't really check correctness, only that get/set are inverses of each other. The problem with primal/dual values is that they're one-way only.

**Comment by odow** [comment]
Merging since this is clearly a win. We can sort the primal/dual and `Variable` bridge tests in other PRs.

File: src/Bridges/Constraint/bridges/geomean.jl (hunk 1 above)
Comment (blegat) [reply src/Bridges/Constraint/bridges/geomean.jl:223]: Nice catch, this is a good validation for the new tests

**Pull request merged**